patches/generativeaipatch.py

```diff
@@ -0,0 +1,24 @@
+import wrapt
+import os
+
+@wrapt.when_imported('google.generativeai')
+def post_import_logic(module):
+    old_configure = module.configure
+    def new_configure(*args, **kwargs):
+        if ('default_metadata' in kwargs):
+            default_metadata = kwargs['default_metadata']
+        else:
+            default_metadata = []
+        kwargs['transport'] = 'rest' # Only support REST requests for now
+        default_metadata.append(("x-kaggle-proxy-data", os.environ['KAGGLE_DATA_PROXY_TOKEN']))
+        default_metadata.append(('x-kaggle-authorization', f'Bearer {os.environ['KAGGLE_USER_SECRETS_TOKEN']}'))
+        kwargs['default_metadata'] = default_metadata
+        if ('client_options' in kwargs):
+            client_options = kwargs['client_options']
+        else:
+            client_options = {}
+        client_options['api_endpoint'] = os.environ['KAGGLE_DATA_PROXY_URL'] + '/palmapi'
```

Comment:
How does that work with other models?

Reply:
All requests for all models will come from the same Python package, we add this prefix to the path so that the data-proxy knows which handler to call. We can change the prefix to be more intuitive if that will help.